extensions/amp-story-360/0.1/amp-story-360.js

```diff
@@ -411,6 +438,19 @@ export class AmpStory360 extends AMP.BaseElement {
     }
   }
 
+  /** @private */
+  restoreOrLoseGlContext_() {
+    if (this.renderer_) {
+      if (this.distance_ < MIN_WEBGL_DISTANCE) {
+        if (this.renderer_.gl.isContextLost()) {
+          this.lostGlContext_.restoreContext();
+        }
+      } else if (!this.renderer_.gl.isContextLost()) {
```

Comment:
Shouldn't we always call `loseContext` when distance > MIN_WEBGL_DISTANCE, to free up memory on systems that support `loseContext`?

Reply:
We only call `loseContext` when distance > MIN_WEBGL_DISTANCE if the context has already been lost.
Otherwise we get an INVALID_OPERATION error:
<img width="478" alt="Screen Shot 2020-11-04 at 9 33 57 AM" src="https://user-images.githubusercontent.com/3860311/98124210-e154a480-1e80-11eb-86bf-cb6e53eee0ec.png">
We do the same check for `restoreContext` too.